### PR TITLE
Issue #109: Update migration guide

### PR DESCRIPTION
**Pull request by sam-byng**
https://github.com/apache/incubator-pekko/issues/109: Update migration guide based on experience working with issue #243

### CODE REVIEW

**Review by pjfanning — CHANGES_REQUESTED** [review]
No - please do not start mentioning poms and SBT stuff. Users should educate themselves on their preferred build tools. Pekko is neutral and should not complicate our docs with pom trivialities.

File: docs/src/main/paradox/project/migration-guides.md [hunk 1]
```diff
@@ -10,16 +10,29 @@ please [migrate to Akka 2.6](https://doc.akka.io/docs/akka/current/project/migra
 
 This is just stub documentation. It will be improved.
 
-* for Pekko jar dependencies, the groupId is "org.apache.pekko" instead of "com.typesafe.akka"
-* the jar names start with "pekko" instead of "akka" - e.g. pekko-actor_2.13.jar instead of akka-actor_2.13.jar 
+In your dependency management files:
+* Update dependency Akka version to the latest Pekko version.
```
Comment (mdedetrich) [inline docs/src/main/paradox/project/migration-guides.md:14]: Rather than latest Pekko version I would instead say latest version of Pekko 1.0.x (which we claim to be broadly compatible with latest Akka BSL release).

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Just one thing I would mention

File: docs/src/main/paradox/project/migration-guides.md (hunk 1 above)
Comment (sam-byng) [reply docs/src/main/paradox/project/migration-guides.md:14]:
```suggestion
* Update dependency Akka version to the latest version of Pekko 1.0.x.
```

**Comment by kw217** [comment]
Looks like all the comments have been addressed - can this be approved and merged?

**Comment by mdedetrich** [comment]
Checking this now

**Review by mdedetrich — APPROVED** [review]
lgtm

**Comment by pjfanning** [comment]
I must admit to not liking the categorisation of the changes and the duplication of the changes because of the artificial categorisation.

**Comment by mdedetrich** [comment]
> I must admit to not liking the categorisation of the changes and the duplication of the changes because of the artificial categorisation.

Can you rephrase, looks fine to me?

File: docs/src/main/paradox/project/migration-guides.md [hunk 2]
```diff
@@ -10,16 +10,29 @@ please [migrate to Akka 2.6](https://doc.akka.io/docs/akka/current/project/migra
...
+* Update dependency Akka version to the latest version of Pekko `1.0.x`.
+* for Pekko jar dependencies, the groupId is `org.apache.pekko` instead of `com.typesafe.akka` or `com.lightbend.akka.<package_name>`
+* the jar names start with "pekko" instead of "akka" - e.g. pekko-actor_2.13.jar instead of akka-actor_2.13.jar
 * Alpakka equivalent is "pekko-connectors" - e.g. pekko-connectors-kafka_2.13.jar instead of alpakka-kafka_2.13.jar
-* Pekko packages start with "org.apache.pekko" instead of "akka" - e.g. `import org.apache.pekko.actor` instead of `import akka.actor`
-* Where class names have "Akka" in the name, the Pekko ones have "Pekko" - e.g. PekkoException instead of AkkaException
-* Configs in `application.conf` use "pekko" prefix instead of "akka"
+
+In config files:
+* Configs use `pekko` prefix instead of `akka`
```
Comment (pjfanning) [inline docs/src/main/paradox/project/migration-guides.md:20]: why remove the reference to 'application.conf' ?

Comment (mdedetrich) [reply docs/src/main/paradox/project/migration-guides.md:20]: To be more accurate we could use `*.conf` instead since some conf files are `reference.conf` and others are `applciation.conf`.

in fact from checking the codebase it should all be `reference.conf`, after all the idea is that `reference.conf` is whats used for libraries and `application.conf` is whats used for applications.

File: docs/src/main/paradox/project/migration-guides.md [hunk 3]
```diff
@@ -10,16 +10,29 @@ please [migrate to Akka 2.6](https://doc.akka.io/docs/akka/current/project/migra
...
+
+In source code files:
+* Pekko packages start with `org.apache.pekko` instead of `akka` - e.g. `import org.apache.pekko.actor` instead of `import akka.actor`
+* Where class names have "Akka" in the name, the Pekko ones have "Pekko" - e.g. `PekkoException` instead of `AkkaException`
+* Configs use `pekko` prefix instead of `akka`
+
+In Documentation:
```
Comment (pjfanning) [inline docs/src/main/paradox/project/migration-guides.md:27]: I think this applies to migration use case. It's useful advice to people contributing code to Pekko - but it is not really something that someone migrating their company's projects needs to know about.

File: docs/src/main/paradox/project/migration-guides.md [hunk 4]
```diff
@@ -10,16 +10,29 @@ please [migrate to Akka 2.6](https://doc.akka.io/docs/akka/current/project/migra
...
+* Where appropriate, text should refer to `Pekko` instead of `Akka`
+* URLs linking to files in https://doc.akka.io/docs/akka/ should instead point at https://pekko.apache.org/docs/pekko/
+
 * We have changed the default ports used by the pekko-remote module.
```
Comment (pjfanning) [inline docs/src/main/paradox/project/migration-guides.md:31]: this port information is useful but it is now sort of orphaned by the artificial breaking up of the migration info into categories

there are not so many bullet points that we need to split it up like this - especially when we then stick some of the bullet points off in 'no category'

File: docs/src/main/paradox/project/migration-guides.md [hunk 5]
```diff
@@ -10,16 +10,29 @@ please [migrate to Akka 2.6](https://doc.akka.io/docs/akka/current/project/migra
...
-* With @ref:[Classic Remoting](../remoting.md), Akka defaults to 2552, while Pekko defaults to 7355.
-* With @ref:[Artery Remoting](../remoting-artery.md), Akka defaults to 25520, while Pekko defaults to 17355.
+    * With @ref:[Classic Remoting](../remoting.md), Akka defaults to 2552, while Pekko defaults to 7355.
+    * With @ref:[Artery Remoting](../remoting-artery.md), Akka defaults to 25520, while Pekko defaults to 17355.
 
+## TBC:
 We are still investigating the effects of how the package name changes affect the @ref:[Persistence](../persistence.md)
```
Comment (pjfanning) [inline docs/src/main/paradox/project/migration-guides.md:36]: See https://lists.apache.org/thread/v7c08l4bn66o4o2go0h7xzzl8pz0tr6p

I can update this section in an independent PR

File: docs/src/main/paradox/project/migration-guides.md (hunk 2 above)
Comment (pjfanning) [reply docs/src/main/paradox/project/migration-guides.md:20]: it's a migration guide for users migrating their own code bases - they will be updating their own application.conf files

some lib maintainers might have reference.conf files

I suppose, some users use '.json' files for configs as well (because typesafe config supports those too). 

I would still prefer if this section, tries to name some of the more common config files to help out users to know what to look for - imagaine if the Akka based app that is being migrated was written 5 years ago and the original writers have moved on.

Comment (mdedetrich) [reply docs/src/main/paradox/project/migration-guides.md:20]: > it's a migration guide for users migrating their own code bases - they will be updating their own application.conf files

Right but its also relevant to libraries which integrate with Pekko, in which case they would be dealing with `reference.conf` files and not `application.conf`. Not all users are applications.

> I suppose, some users use '.json' files for configs as well (because typesafe config supports those too).

And also `*.prop` files too which afaik are compatible with HOCON format.

> I would still prefer if this section, tries to name some of the more common config files to help out users to know what to look for - imagaine if the Akka based app that is being migrated was written 5 years ago and the original writers have moved on.

In that case the easiest thing would be to search for `akka` prefix in files that are known to hold configurations which would be `*.conf`/`*.prop` and `*.json`.
